src/daft-writers/src/batch.rs

```diff
@@ -33,50 +35,62 @@ impl FileWriter for TargetBatchWriter {
     type Input = Arc<MicroPartition>;
     type Result = Option<Table>;
 
-    fn write(&mut self, input: &Arc<MicroPartition>) -> DaftResult<()> {
+    fn write(&mut self, mut input: Arc<MicroPartition>) -> DaftResult<()> {
         assert!(
             !self.is_closed,
             "Cannot write to a closed TargetBatchWriter"
         );
-        let input = if let Some(leftovers) = self.leftovers.take() {
-            MicroPartition::concat([&leftovers, input])?.into()
-        } else {
-            input.clone()
-        };
+        if input.is_empty() {
+            return Ok(());
+        }
+
+        let mut input_size_bytes = input
+            .size_bytes()?
+            .expect("Micropartitions in target batch writer must be loaded");
+
+        if let Some(leftovers) = self.leftovers.take() {
+            input = MicroPartition::concat([leftovers, input])?.into();
```

Comment:
if we had a bunch of small morsels, we can potentially perform this concat every iteration. Ideally we can just keep a buffer and keep track of the count number of bytes / rows that we have

Reply:
Makes sense, implemented a simple buffer for this.